src/clj_http/conn_mgr.clj

```diff
@@ -232,20 +247,26 @@
       get-custom-strategy-registry))
 
 (defn ^BasicHttpClientConnectionManager make-regular-conn-manager
-  [{:keys [keystore trust-store
+  [{:keys [custom-dns-resolver
+           keystore trust-store
            key-managers trust-managers
            socket-timeout] :as req}]
   (let [conn-manager (cond
                        (or key-managers trust-managers)
-                       (BasicHttpClientConnectionManager. (get-managers-scheme-registry req))
+                       (BasicHttpClientConnectionManager. (get-managers-scheme-registry req)
+                                                          nil nil
+                                                          (when custom-dns-resolver (dns-custom-resolver custom-dns-resolver)))
```

Comment:
Agree, if you have some more concrete changes I would be happy to work on it.

Reply:
Remove the wrapper and directly pass in `dns-resolver` to the constructor. This will allow library users to have much more flexibility with how to define a a custom resolver.